app/models/katello/concerns/operatingsystem_extensions.rb

```diff
@@ -6,33 +6,102 @@ module OperatingsystemExtensions
       REDHAT_ATOMIC_HOST_DISTRO_NAME = "Red Hat Enterprise Linux Atomic Host".freeze
       REDHAT_ATOMIC_HOST_OS = "RedHat_Enterprise_Linux_Atomic_Host".freeze
 
+      DEBIAN_DEFAULT_PROVISIONING_TEMPLATE = "Preseed default %{template_kind_name}".freeze
+      DEBIAN_DEFAULT_PTABLE = "Preseed default".freeze
+
+      SUSE_DEFAULT_PROVISIONING_TEMPLATE = "AutoYaST default %{template_kind_name}".freeze
+      SUSE_DEFAULT_PTABLE = "AutoYaST entire SCSI disk".freeze
+
       included do
         after_create :assign_templates!
         before_create :set_atomic_attributes, :if => proc { |os| os.name == ::Operatingsystem::REDHAT_ATOMIC_HOST_OS }
       end
 
-      def assign_templates!
-        # Automatically assign default templates
-        if self.family == 'Redhat'
-          TemplateKind.all.each do |kind|
-            if name == ::Operatingsystem::REDHAT_ATOMIC_HOST_OS && kind.name == "provision"
-              provisioning_template_name = Setting["katello_default_atomic_provision"]
-            else
-              provisioning_template_name = Setting["katello_default_#{kind.name}"]
-            end
+      def find_related_os(ignored_os)
+        [true, false].each do |with_major|
+          query = Operatingsystem.where(name: self.name).where.not(id: ignored_os)
+          query = query.where(major: self.major) if with_major
+          related_os = query.order(created_at: :desc).last
+          return related_os unless related_os.nil?
+        end
+        nil
+      end
 
-            if (template = ProvisioningTemplate.unscoped.find_by_name(provisioning_template_name))
+      def assign_related_os_templates
+        ignored_os = [self.id]
+        3.times do
+          related_os = find_related_os(ignored_os)
+          return false if related_os.nil?
+
+          unless related_os.provisioning_templates.nil? && related_os.ptables.nil?
```

Comment:
Do you anticipate any cases where one of these is nil and the other isn't?

Would you want to keep searching until finding a single related_os where both of these are non-nil? Or allow these to come each from different matched related_os?

Reply:
yes, it should get an OS on which provisioning templates and partition tables exist.